magpie/services.py

```diff
@@ -672,48 +681,64 @@ class ServiceTHREDDS(ServiceInterface):
         models.File: permissions,
     }
 
+    def get_config(self):
+        # type: () -> ConfigDict
+        cfg = super(ServiceTHREDDS, self).get_config()
+        cfg.setdefault("file_patterns", [r".*.nc"])
+        cfg.setdefault("data_type", {"prefixes": []})
+        if not cfg["data_type"]["prefixes"]:
+            cfg["data_type"]["prefixes"] = ["fileServer", "dodsC", "dap4", "wcs", "wms"]
+        cfg.setdefault("metadata_type", {"prefixes": []})
+        if not cfg["metadata_type"]["prefixes"]:
+            cfg["metadata_type"]["prefixes"] = [None, "catalog", "ncml", "uddc", "iso"]
```

Comment:
You provide a great default configuration, but to avoid a back-compatibility migration we could also push all prefixes to the data type. The responsability of the admin overriding this config with metadata prefixes would then have to set the permissions of his instance accordingly.
Note that I would prefer a proper migration, but understand that it could be complex.

Reply:
Indeed. There is quite a bit of work to move this to the db properly. 
Saving the config the db in itself is somewhat easy, but the problem is about how to handle updating it. 

Example: 
First time the config is provided via config.yml and stored to database. 
Then, if omitted another time, how do we handle this? Reuse the thing from db or flush? How to know what is the intended behavior? The only real method would be to have an API call to update, but then more development to support this.

For now, I think it is safe to assume that the config.yml provided will always be there, so the config should be loaded each time as expected.

Reply:
Hum I'm not sure you understand. I don't want to save the config to the db.
I only point out that for an instance without proper thredds configuration (all of them as of speaking), when we will update magpie, the ``if not cfg[...][...]`` will be true and the "default" behavior will apply. This default behavior, with the current instances, will cause to remove all permissions for the metadata_type [None, "catalog", "ncml", "uddc", "iso"] since no BROWSE permissions exist. 
What I propose here, is a solution to https://github.com/Ouranosinc/Magpie/pull/363#discussion_r512801739. If by default, you put all the prefix in ``cfg["data_type"]["prefixes"]``, then when upgrading magpie, even without BROWSE permissions set, everything will work as before. 
The admin will then be free to configure the browse permission for its users/groups and later configure thredds properly using the config.yml.

Reply:
Although that would work, it somewhat goes against to point of adding the permission specifically for metadata in the first place. Isn't it be better to have a default config that matches as closely as possible "normal" usage? I would argue that if for whatever reason we want to preserve the old method, we should provide the config to do so.

When Magpie is going to be updated to support this specific use case, proper configuration should be provided to it.
I think it is cleaner to duplicate existing READ entries as equivalent BROWSE during migration.

Reply:
Yes I agree, that's why I noted that I would prefer a proper migration. I just tought that it may be hard, but let's do that. Assign BROWSE permissions where READ permissions currently exist.